### PR TITLE
Progress reporting on all migration participants

### PR DESCRIPTION
**Pull request by tinwelint**
also makes sure that the migration progress is visible to the user by
using the user log.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/participant/BaseStoreMigrationParticipant.java [hunk 1]
```diff
@@ -33,28 +33,37 @@
  */
 public class BaseStoreMigrationParticipant implements StoreMigrationParticipant
 {
+    protected final String name;
+
+    public BaseStoreMigrationParticipant( String name )
```
Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/participant/BaseStoreMigrationParticipant.java]: Isn't `NoopStoreMigrationParticipant` a better name for this class?

File: community/kernel/src/test/java/org/neo4j/kernel/impl/storemigration/monitoring/VisibleMigrationProgressMonitorTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    public void shouldReportAllPercentageSteps() throws Exception
+    {
+        // GIVEN
+        AssertableLogProvider logProvider = new AssertableLogProvider();
+        Log log = logProvider.getLog( getClass() );
+        VisibleMigrationProgressMonitor monitor = new VisibleMigrationProgressMonitor( log );
+        monitor.started();
+
+        // WHEN
+        monitorSection( monitor, "First", 100, 40, 25, 23 );
```
Comment (davidegrohmann) [inline community/kernel/src/test/java/org/neo4j/kernel/impl/storemigration/monitoring/VisibleMigrationProgressMonitorTest.java]: What should happen if you report more than the `max` value?

File: community/kernel/src/test/java/org/neo4j/unsafe/impl/batchimport/staging/CoarseBoundedProgressExecutionMonitorTest.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import static org.neo4j.unsafe.impl.batchimport.stats.Keys.done_batches;
+
+public class CoarseBoundedProgressExecutionMonitorTest
+{
+    private final Configuration config = new Configuration.Overridden( Configuration.DEFAULT )
+    {
+        @Override
+        public int batchSize()
+        {
+            return 1;
```
Comment (davidegrohmann) [inline community/kernel/src/test/java/org/neo4j/unsafe/impl/batchimport/staging/CoarseBoundedProgressExecutionMonitorTest.java]: Why not having this parametrized?

**Comment by davidegrohmann** [comment]
@tinwelint several test failures to be fixed

File: community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/participant/BaseStoreMigrationParticipant.java (hunk 1 above)
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/participant/BaseStoreMigrationParticipant.java]: Or AbstractionStoreMigrationParticipant... goes in line with most other such base classes.

Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/participant/BaseStoreMigrationParticipant.java]: sure

**Comment by davidegrohmann** [comment]
@tinwelint a couple of tests still failing

**Pull request merged**